### DISCUSSION TITLE
Migrating from Onenote to Trilium

### DISCUSSION THREAD

**Discussion by maphew** [TriliumNext/Trilium#5910] [General]
In few places we've talked about how to migrate or port content from Microsoft OneNote to Trilium. This discussion thread is to give a central location to coordinate and branch off from. I'll edit this top post wiki style to serve as an index to the various places.

**TLDR:** The fastest, easiest, and most solid means I've found so far is to use the Onemore Addin and it's Archive or Export commands: https://onemoreaddin.com/commands/File%20Commands.htm, and then Trilium's standard import from zip or folder. There are some warts. Page dates and times are not respected for example.

* OneNote sample pack [**ON to Trilium**](https://1drv.ms/o/s!AsAr2gmeADJM5UhAeE-ZPlm3EPC6?e=W6SKd7) - _(link should work with any Microsoft account._ OneNote to Trilium, a notebook holding a variety of pages and content types curated from my personal notes, collected together to use as conversion/migration testing data (or it will be, as I trawl through my horde and lift out representative sorta-shiny bits).
  * [ON to Trilium.zip](https://github.com/user-attachments/files/17551712/ON.to.Trilium.zip) - archive export of the notebook using the Onemore addin. _2024-10-28_
  * Result of importing the onenote sample pack: https://www.maphew.com/share/on-to-trilium _2025_05_13_

* Powershell https://github.com/meichthys/onenote-html-export - worked in 2021 but drift in underlying Powershell versions and MS interop assemblies means downgrading powershell to make work ([ref](https://github.com/meichthys/onenote-html-export/issues/1)). Still useful as research.
  * another powershell example, untested https://passbe.com/2019/bulk-export-onenote-2013-2016-pages-as-html/

* Using Evernote Legacy as a bridge https://github.com/zadam/trilium/issues/1013#issuecomment-1532108903. Contains advice on prepping notes for transfer that are useful no matter what mechanism is used.

* A discusion on handling the noise attached images make in the table of contents: TriliumNext/Notes#527 

* ...

----
Edits:
_2024-10-28_ - ON to Trilium archive export; link to image handling discussion
_2025-04-13_ - ON to Trilium archive import results

**Comment by doobes** [comment]
Just a caveat

Be careful with this process.

I did some exporting from OneNote via OneNoteMDExporter and something in one of the exported files didn't agree with Trilium and corrupted the database.

It took me forever to figure out what was going on.

To be honest, it was a blessing as a lot of the stuff I had in OneNote was OBE (Overcome By Events).

Cheers,

**Reply by maphew** [reply]
thanks for the heads up! Do you have any idea what might have triggered the corruption? I presume not, else you'd have said, but maybe there's a vague hunch. How did the corruption reveal itself? An obvious everything-is-broken event or the scariery you thought it was all fine but then...

So far I've imported a hundred pages or so without (a known) problem. My content is relatively simple though: text, screenshots, and the odd attachment.

**Reply by doobes** [reply]
Well, 

I had just set up a server instance. The first time I synced the server came to a virtual halt.

I rebooted the server and re-synced, same thing. I deleted everything from the Trillium desktop and restarted everything. 

About 1/3 of my original data has been saved, but I saw the folder where it hung. So, again, I deleted everything, imported into Trillium, and then deleted the offending folder from the desktop before synching.

After that it was fine. I have no idea what the height was, and at this point had no interest in digging in any further.

Good luck!

**Comment by maphew** [comment]
A status update: I've imported about a thousand pages into Trilium and have mostly left Onenote behind. There will be 10 to 15% onenote usage indefinitely for those activities that a) work with other people still in Onenote, and b) Onenote is still easier for capturing copy-paste while citing source from webpages, Word, Outlook, etc...

I used the _Onemore Addin_ and it's Archive or Export functions to create html files, and did some cleaining on those zips using a Trilium-py python script I hacked together ([ref](https://github.com/maphew/mhwcode/blob/master/metril/import_onenote_zip.py)), and then imported using Trilium's built-in import.

I cleaned up each note manually. I had spent a few weeks trying to script the cleanup but there were so many edge cases I got demotivated and drowned in the effort. I now think this was for the better anyway -- because in addition cleaning up the formatting I also reviewed and sometimes updated the content. I have some good stuff in there! It was good to refresh and re-enjoy that. And also: some bad stuff that I got great pleasure out of deleting. ; -)

**Reply by doobes** [reply]
Wow.

Sounds tedious.

Glad you got it done, and thanks for documenting your efforts. All too often I'm guilty of not doing that.

Cheers

**Reply by maphew** [reply]
I'm learning to periodically embrace tedium as a powerup instead of energy drain, which is my default reaction by far! Sometimes, in measured doses, it can be good. :) That said, my trail of undocumented learnings is far longer and voluminous than the written up ones. Thanks for the thanks.

**Reply by maphew** [reply]
I should note that the python processing clean up step is not required, the exported Onemore archive imports fine, albeit with extra paragraph spacing and other things that are annoying but not of the I-lost-data type. I've added a no-processing result example in the opening post.
